backend/app/src/main/java/ch/zhaw/studyflow/controllers/CalendarController.java

```diff
@@ -0,0 +1,252 @@
+        final HttpRequest request = context.getRequest();
+        HttpResponse response = context.getRequest().createResponse();
+        return authenticator.handleIfAuthenticated(request, principal -> {
+            Optional<Long> userId = principal.getClaim(CommonClaims.USER_ID).map(Long::valueOf);
+            if (userId.isPresent()) {
+                List<Appointment> appointments = appointmentManager.readAllBy(userId.get(), from, to);
+                response.setResponseBody(JsonContent.writableOf(appointments))
+                        .setStatusCode(HttpStatusCode.OK);
+            } else {
+                response.setStatusCode(HttpStatusCode.BAD_REQUEST);
+            }
+            return response;
+        });
+    }
+
+    @Route(path = "getAppointment")
+    @Endpoint(method = HttpMethod.GET)
+    public HttpResponse getAppointment(RequestContext context) {
+        final HttpRequest request = context.getRequest();
+        HttpResponse response = context.getRequest().createResponse();
+        return authenticator.handleIfAuthenticated(request, principal -> {
+            Optional<Long> userId = principal.getClaim(CommonClaims.USER_ID).map(Long::valueOf);
+            if (userId.isPresent()) {
+                request.getRequestBody()
+                        .flatMap(body -> body.tryRead(Appointment.class))
+                        .flatMap(appointment -> {
+                            Appointment foundAppointment = appointmentManager.read(appointment.getCalendarId(), appointment.getId());
+                            return Optional.of(foundAppointment);
+                        })
+                        .ifPresentOrElse(
+                                appointment -> {
+                                    response.setResponseBody(JsonContent.writableOf(appointment))
+                                            .setStatusCode(HttpStatusCode.OK);
+                                },
+                                () -> {
+                                    response.setStatusCode(HttpStatusCode.BAD_REQUEST);
+                                }
+                        );
+            } else {
+                response.setStatusCode(HttpStatusCode.BAD_REQUEST);
+            }
+            return response;
+        });
+    }
+
+
+    @Route(path = "deleteAppointment")
+    @Endpoint(method = HttpMethod.DELETE)
+    public HttpResponse deleteAppointment(RequestContext context) {
+        final HttpRequest request = context.getRequest();
+        HttpResponse response = request.createResponse();
+        return authenticator.handleIfAuthenticated(request, principal -> {
+            request.getRequestBody()
+                    .flatMap(body -> body.tryRead(Appointment.class))
+                    .ifPresentOrElse(
+                            appointment -> {
+                                appointmentManager.delete(appointment.getId());
```

Comment:
Should we check if the user is allowed to delete the appointment?

Reply:
I think it would make more sense to pass the appointment id to the endpoint through a route parameter.